File: Solutions/17. Longest Palindrome/sol.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int odds = 0;
        for (char c='A'; c<='z'; c++)
            odds += count(s.begin(), s.end(), c) & 1;
        int ans = s.length() - odds;
        if(odds)
            ans++ ;
        return ans ;
    }
};

File: Solutions/17. Longest Palindrome/sol.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        freq = [0]*128
        for c in s:
            freq[ord(c)] += 1
        odd = 0
        for i in freq: odd += i & 1
        return len(s) - odd + (odd > 0) 